plugins/woocommerce/src/Internal/Admin/Logging/FileV2/File.php

```diff
@@ -44,16 +46,38 @@ class File {
 	 */
 	protected $hash = '';
 
+	/**
+	 * The file's resource handle when it is open.
+	 *
+	 * @var resource
+	 */
+	protected $stream;
+
 	/**
 	 * Class File
 	 *
 	 * @param string $path The absolute path of the file.
 	 */
 	public function __construct( $path ) {
+		global $wp_filesystem;
+		if ( ! $wp_filesystem instanceof WP_Filesystem_Direct ) {
+			WP_Filesystem();
```

Comment:
I was wondering if we should handle a case where this returns false or null (imagine a 'rogue' filter on `filesystem_method` or `filesystem_method_file`) ... but perhaps the answer is that there isn't much we could do in such a situation?

Reply:
Good question. `WP_Filesystem()` does return `false` if something goes wrong, so we could capture that, but I'm not sure what we'd do with it... at that point it might be best to just let an error get triggered when we try to access a method on `$wp_filesystem` that doesn't exist.

I added [a card](https://github.com/orgs/woocommerce/projects/184/views/1?pane=issue&itemId=44071475) to the project board to think more about this.